cdap-api/src/main/java/co/cask/cdap/internal/workflow/ProgramWorkflowAction.java

```diff
@@ -42,11 +50,16 @@
   private String programName;
   private Runnable programRunner;
   private SchedulableProgramType programType;
+  private Map<String, Dataset> datasets;
+  private TransactionSystemClient txClient;
 
-  public ProgramWorkflowAction(String name, String programName, SchedulableProgramType programType) {
+  public ProgramWorkflowAction(String name, String programName, SchedulableProgramType programType,
+                               @Nullable TransactionSystemClient txClient) {
```

Comment:
This is nullable because in WorkflowDriver's executeAction, we only need to pass in the txClient only if we are doing a custom action